### ISSUE TITLE
Question: Slow build times with SA1027 enabled?

### ISSUE THREAD

**Issue by CodeBlanch** [DotNetAnalyzers/StyleCopAnalyzers#3125]
I just enabled StyleCop in a solution that has existed forever without any kind of style standards.

With...
```xml
<Rule Id="SA1027" Action="None" /> <!-- Use tabs not spaces -->
```

Build takes ~13 seconds.

With...
```xml
<Rule Id="SA1027" Action="Warning" /> <!-- Use tabs not spaces -->
```

Build takes ~7.5 **minutes**. 136,562 SA1027 warnings generated.

It looks like for each file using spaces SA1027 generates a warning on every line. Is that really how SA1027 should work? Maybe one warning per file would be better?

**Comment by sharwell** [comment]
This is a known issue in the compiler (the evaluation process for whether a diagnostic should be shown or not is very slow, so a moderate number of diagnostics like seen above is slow in practice).

**Comment by sharwell** [comment]
This is more a performance issue for dotnet/roslyn than an issue here. The compiler is expected to perform adequately even if 100,000+ warnings are reported.

**Comment by CodeBlanch** [comment]
@sharwell Thanks for explaining the source of the slowdown! Still think the rule might be woth revisiting. Let's pretend the compiler(?) could magically show 136,562 CA1027 warnings instantly. Great, but is that particularly useful to anyone? 😄 I had to turn off SA1027 in this solution. If I could have made it one warning per file, or somehow tell it to max out, I would have left it on. Just some feedback for you to consider in the future (beyond the slowdown issue).
